content/languages/c/authoring/memory-management-techniques.md

```diff
@@ -0,0 +1,24 @@
+---
+kind: tutorial
+languages: [c]
+sidebar: "language:c"
+---
+
+# Memory Management in C kata
+
+_TBD_
+
+## Arrays and strings
+
+- malloc in solution and free in tests
+- pass in a preallocated buffer (use size hints if possible)
+- two functions: get size, allocate in tests, run solution
+- two functions: solution with allocation, deallocation. Bookkeeping information managed by user or passed as additional `void*`
+- one function: accept buffer+size, return retsult or error and required size
+- avoid string constants, use named symbols
+- reporting size:output param, structure, sentinel teminators
+## Two-dimensional arrays
+
+- N+1 allocations
+- Flat
+- TOC + Flat
```

Comment:
TOC? I don't know or don't remember what it means...

Reply:
"TOC" as in "table of contents", I made the name up and I do not know how (if) it is professionally called: two malloced buffers, one flat array of `T*` and another one of `T**` with elements pointing to "rows" in the first one.

This part is not ready yet, but if you'd like to discuss it, please use dedicated thread #230 or PR #231